packages/sdk/src/wallet-utils/validate-tx.js

```diff
@@ -7,21 +8,43 @@ import {invariant} from "@onflow/util-invariant"
 const isPayer = signable => {
   return signable.roles.payer
 }
+
 const getVoucher = signable => {
   return signable.voucher
 }
+
 const getMessage = signable => {
   return signable.message
 }
 
+const getDomainTag = signable => {
+  return getMessage(signable).substring(0, TRANSACTION_DOMAIN_TAG.length)
+}
+
 const isExpectedMessage = signable => {
   return isPayer(signable)
     ? encodeOutsideMessage(getVoucher(signable)) === getMessage(signable)
     : encodeInsideMessage(getVoucher(signable)) === getMessage(signable)
 }
```

Comment:
I think the newer direction for this stuff is can we get what needs to be signed via the signable and the voucher
Which means, the naming here is now off, but the logic is roughly the same but less
```javascript
const isExpectedMessage = signable => {
  return isPayer(signable)
    ? encodeOutsideMessage(getVoucher(signable))
    : encodeInsideMessage(getVoucher(signable))
}
```